include/seqan3/io/structure_file/output.hpp

```diff
@@ -0,0 +1,967 @@
+                        options,
+                        seq,
+                        id,
+                        bpp,
+                        structure,
+                        energy,
+                        react,
+                        react_error,
+                        comment,
+                        offset);
+            }
+        }, format);
+    }
+
+    //!\brief Write columns to file format, only tag-dispatch once.
+    template <input_range_concept seq_type,
+              input_range_concept id_type,
+              input_range_concept bpp_type,
+              input_range_concept structure_type,
+              input_range_concept structured_seq_type,
+              input_range_concept energy_type,
+              input_range_concept react_type,
+              input_range_concept comment_type,
+              input_range_concept offset_type>
+    void write_columns(seq_type && seq,
+                       id_type && id,
+                       bpp_type && bpp,
+                       structure_type && structure,
+                       structured_seq_type && structured_seq,
+                       energy_type && energy,
+                       react_type && react,
+                       react_type && react_error,
+                       comment_type && comment,
+                       offset_type && offset)
+    {
+        static_assert(!(detail::decays_to_ignore_v<reference_t<seq_type>> &&
+                        detail::decays_to_ignore_v<reference_t<id_type>> &&
+                        detail::decays_to_ignore_v<reference_t<bpp_type>> &&
+                        detail::decays_to_ignore_v<reference_t<structure_type>> &&
+                        detail::decays_to_ignore_v<reference_t<structured_seq_type>> &&
+                        detail::decays_to_ignore_v<reference_t<energy_type>> &&
+                        detail::decays_to_ignore_v<reference_t<react_type>> &&
+                        detail::decays_to_ignore_v<reference_t<comment_type>> &&
+                        detail::decays_to_ignore_v<reference_t<offset_type>>),
+                      "At least one of the columns must not be set to std::ignore.");
+
+        static_assert(detail::decays_to_ignore_v<reference_t<structured_seq_type>> ||
+                      (detail::decays_to_ignore_v<reference_t<seq_type>> &&
+                       detail::decays_to_ignore_v<reference_t<structure_type>>),
+                      "You may not select field::STRUCTURED_SEQ and either of field::SEQ and field::STRUCTURE "
+                      "at the same time.");
+
+        assert(!format.valueless_by_exception());
+        std::visit([&] (auto & f)
+        {
+            if constexpr (!detail::decays_to_ignore_v<reference_t<structured_seq_type>>)
+            {
+                auto zipped = ranges::view::zip(structured_seq, id, bpp, energy, react, react_error, comment, offset);
+
+                for (auto && v : zipped)
+                    f.write(stream,
```

Comment:
please wrap in `{ }` when the statement spans multiple lines even if it's just one statement.